OCCommunicationLib/OCCommunicationLib/OCCommunication.m

```diff
@@ -135,60 +135,69 @@ -(id) initWithUploadSessionManager:(AFURLSessionManager *) uploadSessionManager
     self = [super init];
     
     if (self) {
-        
-        //Init the Queue Array
-        _uploadOperationQueueArray = [NSMutableArray new];
-        
+    
         //Init the Donwload queue array
-        _downloadOperationQueueArray = [NSMutableArray new];
+        self.downloadTaskNetworkQueueArray = [NSMutableArray new];
         
         //Credentials not set yet
-        _kindOfCredential = credentialNotSet;
+        self.kindOfCredential = credentialNotSet;
         
-        //Network Queue
-        _networkOperationsQueue =[NSOperationQueue new];
-        [_networkOperationsQueue setMaxConcurrentOperationCount:NSOperationQueueDefaultMaxConcurrentOperationCount];
+        [self setSecurityPolicyManagers:[self createSecurityPolicy]];
         
-        [self setSecurityPolicy:[self createSecurityPolicy]];
+        self.uploadSessionManager = uploadSessionManager;
+        self.downloadSessionManager = downloadSessionManager;
         
-        _uploadSessionManager = uploadSessionManager;
-        _downloadSessionManager = downloadSessionManager;
+        [self initUploadAndDownloadNoBackgroundManagers];
     }
     
     return self;
 }
 
+- (void) initUploadAndDownloadNoBackgroundManagers {
+    
+    NSURLSessionConfiguration *uploadNoBackgroundConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
```

Comment:
Yes, we could use the same configuration but I think that having both configuration separately it is better to be clear.